app/grandchallenge/evaluation/models.py

```diff
@@ -583,6 +581,12 @@ def save(self, *args, **kwargs):
 
         if adding:
             self.assign_permissions()
+            followers = list(self.phase.challenge.get_admins())
+            followers.append(self.creator)
+            for user in followers:
+                follow(
+                    user=user, obj=self, actor_only=False, send_action=False,
```

Comment:
Does this create 1 follow per submission? Some challenges have tens of thousands of submissions, to unsubscribe from these notifications would the user have to unsubscribe from each one?

Reply:
Oh, hm, yes this creates 1 follow per submission and users would have to unsubscribe from each one (although selecting multiple follows and bulk delete is possible). I'm not sure if unsubscribing and showing these follows to users makes sense though. Maybe it's better to hide them from the subscription list view. Ultimately they are just used to track the status of the submission and to inform users of status changes.

Reply:
If I see it correctly, the status only gets updated to failed or succesful once per submission, correct? In that case, we could also delete the follow after the status change (and after the action/notification has been sent).

Reply:
Yes - the state actually comes from the evaluation that is foreign keyed to the submission, you can have many evaluations per submission but this is rare. The use case that must be implemented here is: as a challenge administrator, I want to opt in and opt out to all evaluation notifications about a particular phase that I am an administrator for. So, if there is a new evaluation, this admin should not be notified if they have opted out. I would have thought that following a phase could do this.

Reply:
For admins, that's actually a really good idea, but that wouldn't work for the submission creator. The submission creator should only be notified about status updates for their own submissions. So then for the users we'd still need the submission follows, or am I missing something?

Reply:
A solution would be to adjust the notification creation signal and have a special rule for when the action target is a phase (send to admins of challenge and creator of submission). It's not as elegant though and would make the notification creation signal a bit less intuitive yet. But we'd have way less follow objects.

Reply:
I pushed the alternative implemention with Phase follows. This is still missing a management command to make all existing challenge admins followers of existing challenge phases, but otherwise, this is what it could look like if we use phase rather than submission follows.